### PR TITLE
Add Azure blob storage source

### PR DESCRIPTION
**Pull request by nicoloboschi**
Changes:
* New source `azure-blob-storage-source` that reads azure blobs
* The auth properties are the same as the code storage config
* The properties are idle-time and file-extensions like the s3-source

### CODE REVIEW

File: examples/applications/http-request-processor/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Http Request processor
```
Comment (eolivelli) [inline examples/applications/http-request-processor/README.md:1]: This seems unrelated

File: langstream-agents/langstream-agent-azure-blob-storage-source/src/main/java/ai/langstream/agents/azureblobstorage/source/AzureBlobStorageSource.java [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+        }
+
+        @Override
+        public String origin() {
+            return null;
+        }
+
+        @Override
+        public Long timestamp() {
+            return System.currentTimeMillis();
```
Comment (eolivelli) [inline langstream-agents/langstream-agent-azure-blob-storage-source/src/main/java/ai/langstream/agents/azureblobstorage/source/AzureBlobStorageSource.java:229]: This should be set at instantiation time, otherwise different calls with return different results

**Review by eolivelli — APPROVED** [review]
The azure part lgtm
But you still have some code from the http request processor

File: examples/applications/http-request-processor/README.md (hunk 1 above)
Comment (nicoloboschi) [reply examples/applications/http-request-processor/README.md:1]: yes it's based on the http-request branch, Ill put this in draft until we get the other one merged

File: langstream-agents/langstream-agent-azure-blob-storage-source/src/main/java/ai/langstream/agents/azureblobstorage/source/AzureBlobStorageSource.java (hunk 2 above)
Comment (nicoloboschi) [reply langstream-agents/langstream-agent-azure-blob-storage-source/src/main/java/ai/langstream/agents/azureblobstorage/source/AzureBlobStorageSource.java:229]: interesting, I copied it from S3 source
